js/main.js

```diff
@@ -15,3 +80,105 @@ navItem.forEach((n) => {
     document.body.classList.remove('hidden-flow');
   });
 });
+
+const createProject = (project) => {
+  const projectEl = document.createElement('div');
+  projectEl.classList.add('project');
+
+  const img = document.createElement('img');
+  img.src = project.featuredImage;
+
+  const name = document.createElement('h3');
+  name.classList.add('project-heading');
+  name.innerText = project.name;
+
+  const ul = document.createElement('ul');
+  ul.classList.add('techs');
+
+  project.technologies.forEach((el) => {
+    const li = document.createElement('li');
+    li.classList.add('techs-item');
+
+    li.innerText = el;
+    ul.appendChild(li);
+  });
+
+  const btn = document.createElement('button');
+  btn.classList = 'btn see-project';
+  btn.id = project.id;
+  btn.innerText = 'See project';
+
+  projectEl.appendChild(img);
+  projectEl.appendChild(name);
+  projectEl.appendChild(ul);
+  projectEl.appendChild(btn);
+
+  return projectEl;
+};
+
+// const modalContainer = document.querySelector('.modal-container')
+const renderPopup = (project) => `
+
+  <div class="img-box">
+    <img
+    src='${project.featuredImage}'
+    class="popup-img"
+    alt=""
+    />  
+  </div>
+
+  <h5>${project.name}</h5>
+  
+  <ul class="techs">
+    <li class="techs-item">Ruby on Rails</li>
+    <li class="techs-item">CSS</li>
+    <li class="techs-item">JavaScript</li>
+  </ul>
+
+  <p>
+    ${project.description}
+  </p>
+
+  <ul>
+    <li>
+      <a href="${project.liveLink}" class="btn">
+        See live
+        <img src="./images/Icon - Export.svg" alt="" />
+      </a>
+    </li>
+    <li>
+      <a href="${project.sourceLink}" class="btn">
+        See source
+        <img src="./images/Icon -GitHub.svg" alt="" />
+      </a>
+    </li>
+  </ul>
+
+`;
+
+window.onload = () => {
+  projects.forEach((project) => {
+    projectsContainer.appendChild(createProject(project));
+  });
+
+  const projectBtns = document.querySelectorAll('.see-project');
+  projectBtns.forEach((btn) => {
+    btn.addEventListener('click', () => {
+      const project = projects.find((project) => project.id === +btn.id);
+      modalContent.innerHTML = renderPopup(project);
+      modal.style.display = 'block';
+    });
+  });
+};
+
+document.addEventListener('DOMContentLoaded', () => {
+  close.onclick = () => {
+    modal.style.display = 'none';
+  };
+
+  window.onclick = (e) => {
+    if (e.target === modal) {
+      modal.style.display = 'none';
+    }
+  };
+});
```

Comment:
- [ ] Nice effort :+1: but you are code is not `DRY` kindly use `DRY` principles. `DRY` stand for `Don't Repeat Yourself`, a basic principle of software development aimed at reducing the repetition of information. For more information go to this [link](https://dev.to/richardwynn/dry-principle-in-100-seconds-2l8c) and export it :slightly_smiling_face:  Also check my suggestion below :point_down: 

```suggestion
const createElement = (tag, className) => {
  const el = document.createElement(tag);
  el.classList.add(className);
  return el;
};

const createList = (items, className) => {
  const ul = createElement('ul', className);

  items.forEach((item) => {
    const li = createElement('li', `${className}-item`);
    li.innerText = item;
    ul.appendChild(li);
  });

  return ul;
};

const createButton = (className, id, text) => {
  const btn = createElement('button', className);
  btn.classList = 'btn see-project';
  btn.id = id;
  btn.innerText = text;
  return btn;
};

const createProject = (project) => {
  const projectEl = createElement('div', 'project');

  const img = createElement('img', 'project-img');
  img.src = project.featuredImage;
  projectEl.appendChild(img);

  const name = createElement('h3', 'project-heading');
  name.innerText = project.name;
  projectEl.appendChild(name);

  const techs = createList(project.technologies, 'techs');
  projectEl.appendChild(techs);

  const btn = createButton('see-project', project.id, 'See project');
  projectEl.appendChild(btn);

  return projectEl;
};

const renderPopup = (project) => {
  const techs = createList(project.technologies, 'techs');

  return `
    <div class="img-box">
      <img src='${project.featuredImage}' class="popup-img" alt="" />
    </div>
    <h5>${project.name}</h5>
    ${techs.outerHTML}
    <p>${project.description}</p>
    <ul>
      <li>
        <a href="${project.liveLink}" class="btn">
          See live
          <img src="./images/Icon - Export.svg" alt="" />
        </a>
      </li>
      <li>
        <a href="${project.sourceLink}" class="btn">
          See source
          <img src="./images/Icon -GitHub.svg" alt="" />
        </a>
      </li>
    </ul>
  `;
};

const openModal = (project) => {
  modalContent.innerHTML = renderPopup(project);
  modal.style.display = 'block';
};

window.onload = () => {
  projects.forEach((project) => {
    const projectEl = createProject(project);
    projectEl.querySelector('.see-project').addEventListener('click', () => openModal(project));
    projectsContainer.appendChild(projectEl);
  });
};

document.addEventListener('DOMContentLoaded', () => {
  close.onclick = () => {
    modal.style.display = 'none';
  };

  window.onclick = (e) => {
    if (e.target === modal) {
      modal.style.display = 'none';
    }
  };
});

```

**Here are the changes that were made:**

- Created helper functions `createElement`, `createList`, and `createButton` to remove repetitive code.
- Changed the structure of the `createProject` function to use the helper functions and avoid repeating code.
- Moved the `renderPopup` function into its own function to make it easier to read and edit.
- Created a separate `openModal` function to handle opening the modal with the correct project data.
- Moved the event listener for the "See project" button inside the `window.onload` event listener to avoid attaching the listener before the DOM is loaded.